discord_slash/client.py

```diff
@@ -349,15 +337,23 @@ def add_slash_command(self,
         """
         name = name or cmd.__name__
         name = name.lower()
-        guild_ids = guild_ids if guild_ids else []
-        if name in self.commands:
-            tgt = self.commands[name]
-            if not tgt.has_subcommands:
-                raise error.DuplicateCommand(name)
-            has_subcommands = tgt.has_subcommands
-            for x in tgt.allowed_guild_ids:
-                if x not in guild_ids:
-                    guild_ids.append(x)
+        guild_ids = guild_ids if guild_ids else ["global"]
+
+        for i in range(len(guild_ids)):
```

Comment:
I chose to use `i in range` as I'm updating the list of which I'm iterating over

Reply:
Pretty certain that that won't work, because the length of the list is got once at the start of the loop, and therefore if you append anything to it the for loop won't get to there, because it's still only iterating the length of the original list.
Tested this with this:
```py
x = [1,2,3,4,5]
for i in range(len(x)):
    print(x[i])
    x.append(i+5)
print(x)
``` And yeah it doesn't work

Reply:
It is at this moment I realised that I'm following rules for another language. I forgot python can handle modifying the list you're iterating over XD